acmw-db-app/pages/forgotPassword.js

```diff
@@ -0,0 +1,89 @@
+import Head from 'next/head'
+import styles from '../styles/Reset.module.css'
+import React, { useState } from 'react';
+  
+class forgotPassword extends React.Component {
+    constructor() {
+    super();
+    this.state = {
+      input: {},
+      errors: {}
+    };
+     
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+     
+  handleChange(event) {
+    let input = this.state.input;
+    input[event.target.name] = event.target.value;
+  
+    this.setState({
+      input
+    });
+  }
+     
+  handleSubmit(event) {
+    event.preventDefault();
+  
+    if(this.validate()){
+        let input = {};
+        input["email"] = "";
+        this.setState({input:input});
+    }
+  }
+  
+  validate(){
+      let input = this.state.input;
+      let errors = {};
+      let isValid = true;
+
+       if ((typeof input["email"] !== "") || (typeof input["email"] !== "undefined") || (typeof input["email"] !== null)) {
```

Comment:
Logic out of order (if it's undefined you cannot check the value so check for undefined first), needs && not ||, and "" and null are not types: `if (typeof(input["email"]) !== "undefined" && input["email"] !== null && input["email"] !== "") {`

Reply:
This line still needs to look what I said in the comment above